drivers/input/input_esp32_touch_sensor.c

```diff
@@ -0,0 +1,390 @@
+/*
+ * Copyright (c) 2023 Espressif Systems (Shanghai) Co., Ltd.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT espressif_touch_sensor
+
+#include <zephyr/device.h>
+#include <zephyr/input/input.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include <esp_err.h>
+#include <soc/soc_pins.h>
+#include <soc/periph_defs.h>
+#include <hal/touch_sensor_types.h>
+#include <hal/touch_sensor_hal.h>
+#include <driver/rtc_io.h>
+#include <esp_intr_alloc.h>
+
+LOG_MODULE_REGISTER(espressif_touch_sensor, CONFIG_INPUT_LOG_LEVEL);
+
+#if defined(CONFIG_COUNTER_RTC_ESP32)
+#error "Conflict detected: COUNTER_RTC_ESP32 enabled"
+#endif /* defined(CONFIG_COUNTER_RTC_ESP32) */
```

Comment:
@fabiobaltieri depends on !COUNTER_RTC_ESP32 was removed from `[drivers/input/Kconfig.esp32](https://github.com/zephyrproject-rtos/zephyr/pull/64931/files#diff-2a08943e5d685a05a0c379c76b8e095b3f0d4835791d6749c9368c8429785991)
and this snippet was inserted here instead.

Reply:
Good stuff, this is ok but you can write it in a more compact way using `BUILD_ASSERT`:

```c
BUILD_ASSERT(!IS_ENABLED(CONFIG_COUNTER_RTC_ESP32),
             "Conflict detected: COUNTER_RTC_ESP32 enabled");

```

up to you

Reply:
Thank you for the good suggestion